src/services/naming-services/EnsNamingService.ts

```diff
@@ -74,9 +97,94 @@ export class EnsNamingService extends NamingService {
 
   //#region private methods
 
+  //////////////////////
+  // decode records   //
+  //////////////////////
+
+  private decodeContentHash(contentHash: string): string {
+    try {
+      const codec = EnsContentHash.getCodec(contentHash);
+      let decodedValue = EnsContentHash.decode(contentHash);
+      if (codec === 'ipfs') {
+        decodedValue = EnsContentHash.cidV0ToV1Base32(decodedValue);
+      }
+
+      return `/${codec}/${decodedValue}`;
+    } catch (error) {
+      return contentHash;
+    }
+  }
+
+  private decodeAddr(key: string, value: string): string {
+    try {
+      const keyComponents = key.split('.');
+      if (keyComponents.length !== 2) {
+        return value;
+      }
+
+      const coinType = parseInt(keyComponents[1], 10);
+      if (!_.isInteger(coinType)) {
+        return value;
+      }
+      const format = EnsAddressEncoder.formatsByCoinType[coinType];
+      const bufferValue = prepareHexValue(value);
+      return format.encoder(Buffer.from(bufferValue, 'hex'));
+    } catch (error) {
+      return value;
+    }
+  }
+
+  private decodeAbi(
+    key: string,
+    value: string,
+  ): { decodedKey: string; decodedValue: string } {
+    try {
+      const keyComponents = key.split('.');
+      // ABI can have multiple formats, we only support plain JSON which is 1
+      // https://docs.ens.domains/ens-improvement-proposals/ensip-4-support-for-contract-abis
+      if (keyComponents.length !== 2 || keyComponents[1] !== '0x01') {
+        return {
+          decodedKey: FabricatedRecordsTypes.AbiUnknown,
+          decodedValue: value,
+        };
+      }
+      const bufferValue = prepareHexValue(value);
+      const jsonString = Buffer.from(bufferValue, 'hex').toString('utf8');
+      return {
+        decodedKey: FabricatedRecordsTypes.AbiJson,
+        decodedValue: JSON.parse(jsonString),
```

Comment:
The return type of JSON.parse is no longer a string, right? So we should either update the return type of the function, or do
```suggestion
        decodedValue: JSON.stringify(JSON.parse(jsonString)),
```